mailpoet/tests/integration/Segments/DynamicSegments/Filters/WooCommerceAverageSpentTest.php

```diff
@@ -0,0 +1,132 @@
+    $this->assertEqualsCanonicalizing(['2@e.com'], $matchingEmails);
+    $matchingEmails = $this->getMatchingEmails('<', 0.01);
+    $this->assertEqualsCanonicalizing([], $matchingEmails);
+    $matchingEmails = $this->getMatchingEmails('<', 736.68);
+    $this->assertEqualsCanonicalizing(['1@e.com', '2@e.com', '3@e.com'], $matchingEmails);
+  }
+
+  public function testItWorksWithAmountNotEqual(): void {
+    $this->createCustomerWithOrderValues('1@e.com', [1, 2, 3, 4]); // average = 2.50
+    $this->createCustomerWithOrderValues('2@e.com', [22.22, 33.33, 44.44]); // average = 33.33
+    $this->createCustomerWithOrderValues('3@e.com', [30000, 30000, 30000]); // big spender!
+
+    $matchingEmails = $this->getMatchingEmails('!=', 2.50);
+    $this->assertEqualsCanonicalizing(['2@e.com', '3@e.com'], $matchingEmails);
+    $matchingEmails = $this->getMatchingEmails('!=', 33.33);
+    $this->assertEqualsCanonicalizing(['1@e.com', '3@e.com'], $matchingEmails);
+    $matchingEmails = $this->getMatchingEmails('!=', 30000);
+    $this->assertEqualsCanonicalizing(['1@e.com', '2@e.com'], $matchingEmails);
+    $matchingEmails = $this->getMatchingEmails('!=', 8);
+    $this->assertEqualsCanonicalizing(['1@e.com', '2@e.com', '3@e.com'], $matchingEmails);
+  }
+
+  public function testItWorksWithDateRanges(): void {
+    $id1 = $this->tester->createCustomer('1@e.com');
+    $this->createOrder($id1, 100, 3);
+    $this->createOrder($id1, 200, 6); // 150 average
+    $this->createOrder($id1, 60, 9); // 120 average
+    $this->createOrder($id1, 400, 100); // 190 average
+
+    $emails = $this->getMatchingEmails('=', 100, 3);
+    $this->assertEqualsCanonicalizing(['1@e.com'], $emails);
+    $emails = $this->getMatchingEmails('=', 100, 6);
+    $this->assertEqualsCanonicalizing([], $emails);
+    $emails = $this->getMatchingEmails('=', 150, 6);
+    $this->assertEqualsCanonicalizing(['1@e.com'], $emails);
+    $emails = $this->getMatchingEmails('=', 120, 9);
+    $this->assertEqualsCanonicalizing(['1@e.com'], $emails);
+    $emails = $this->getMatchingEmails('=', 120, 99);
+    $this->assertEqualsCanonicalizing(['1@e.com'], $emails);
+    $emails = $this->getMatchingEmails('=', 190, 100);
+    $this->assertEqualsCanonicalizing(['1@e.com'], $emails);
+  }
+
+  private function getMatchingEmails(string $operator, float $amount, int $days = 365): array {
+    $filterData = new DynamicSegmentFilterData(DynamicSegmentFilterData::TYPE_WOOCOMMERCE, WooCommerceAverageSpent::ACTION, [
+      'operator' => $operator,
+      'amount' => $amount,
+      'days' => $days,
+    ]);
+
+    return $this->tester->getSubscriberEmailsMatchingDynamicFilter($filterData, $this->averageSpentFilter);
+  }
+
+  private function createCustomerWithOrderValues(string $customerEmail, array $values): void {
+    $customerId = $this->tester->createCustomer($customerEmail);
+    foreach ($values as $value) {
+      $this->createOrder($customerId, $value );
```

Comment:
I'm a bit torn on how to approach this. Can we easily do everything we need to using just `\MailPoet\WooCommerce\Helper::wcCreateOrder()`? That function is just a wrapper for `wc_create_order`, which as best as I can tell doesn't accept arguments for creation date or total, so at some point we would need to make changes to the returned `WC_Order` object. 

We're [already doing that](https://github.com/mailpoet/mailpoet/blob/fc6ab08ccdbf6422c9a947a30086fda3abfc1e8c/mailpoet/tests/_support/IntegrationTester.php#L104-L110) for `date_created` and `billing_email` in our helper method. I've updated the helper to also accept a `total`, and to pass the other arguments through to `wcCreateOrder` so we can pass default arguments like `status` and `customer_id`. This means not having to call `updateWooOrderStats` in this `createOrder`. Does that address your primary concern?

I still like having a `createOrder` helper here because I call it in a few places and wouldn't want to have to repeat everything each time.

Please do let me know if you can think of a more elegant solution, which I'm sure exists.